UnityProject/Assets/Scripts/Items/Others/RandomItemSpot.cs

```diff
@@ -39,68 +42,58 @@ public void OnSpawnServer(SpawnInfo info)
 		public void RollRandomPool(bool UnrestrictedAndspawn, bool overrideTrigger = false)
 		{
 			if (overrideTrigger == false && triggerManually) return;
+			SpawnRandomItems(UnrestrictedAndspawn);
+			if (deleteAfterUse)
+			{
+				_ = Despawn.ServerSingle(gameObject);
+				return;
+			}
+			ShoveIntoStorage(gameObject);
+			APCtoSet.OrNull()?.RemoveFromAPC();
+		}
 
-			var RegisterTile = this.GetComponent<RegisterTile>();
+		[Button]
+		public void SpawnRandomItems(bool unrestrictedAndSpawn = true)
+		{
+			var spawnAmount = 0;
 			for (int i = 0; i < lootCount; i++)
 			{
-				PoolData pool = null;
-				// Roll attempt is a safe check in the case the mapper did tables with low % and we can't find
-				// anything to spawn in 5 attempts
-				int rollAttempt = 0;
-
-				while (pool == null)
-				{
-					if (rollAttempt >= MaxAmountRolls)
-					{
-						break;
-					}
-
-					var tryPool = poolList.PickRandom();
-					if (DMMath.Prob(tryPool.Probability))
-					{
-						pool = tryPool;
-					}
-					else
-					{
-						rollAttempt++;
-					}
-				}
-
-				if (pool == null)
-				{
-					continue;
-				}
-
-				GenerateItem(pool, UnrestrictedAndspawn);
-				if (UnrestrictedAndspawn == false) return;
+				PoolData pool = GrabRandomPool();
+				if (pool == null) continue;
+				GenerateItem(pool, unrestrictedAndSpawn);
+				spawnAmount++;
+				if (unrestrictedAndSpawn == false) break; // what the fuck does UnrestrictedAndspawn mean? What's the point? terrible name
```

Comment:
I don't remember doing this unrestricted thing, possibly was Bod trying to fix a bug?

Reply:
I'm not sure what it is trying to fix or do, so I'm leaving it as is for the time being.

Reply:
https://github.com/unitystation/unitystation/pull/7492

Reply:
I'll leave it as is for now